sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/view.scala

```diff
@@ -35,6 +35,56 @@ object EliminateView extends Rule[LogicalPlan] with CastSupport {
   }
 }
 
+/**
+ * ViewBindingMode is used to specify the expected schema binding mode when we want to create or
+ * replace a view in [[CreateViewStatement]].
+ */
+sealed trait ViewSchemaMode {
+  override val toString: String = getClass.getSimpleName.stripSuffix("$")
```

Comment:
We don't need this override since all subclasses of `ViewSchemaMode` overrides `toString`

Reply:
Intelij seems to disagree
<img width="553" alt="Screenshot 2024-05-09 at 9 13 37 AM" src="https://github.com/apache/spark/assets/3514644/4af555f4-742d-4a86-a605-02ba14ee129c">

Reply:
I mean remove this entire line, not just the keyword `override`